riscv/src/compiler.rs

```diff
@@ -73,6 +73,125 @@ machine Binary(latch, function_id) {
         {function_id', A_byte, B_byte, C_byte} in {P_operation, P_A, P_B, P_C};
     }
 }
+"#,
+        r#"
+// Implements the poseidon permutation for the BN254 curve.
+// Note that this relies on the trace table being non-wrapping, so it will
+// only work with the Halo2 backend (which is the only backend that supports
+// the BN254 curve).
```

Comment:
@georgwiese just mentioned to me that this comment is outdated, leaving a comment as a reminder